### PR TITLE
Configured checkstyle with gradle and travis  #11

### PR DESCRIPTION
**Pull request by OlliKiljunen**
Using checkstyle with Google Java Style (google_checks). One can check their coding style with `./gradlew check`. The same check is done by Travis and build fails if there are style issues.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -134,4 +134,8 @@ gradle-app.setting
 ### Gradle Patch ###
 **/build/
 
-# End of https://www.gitignore.io/api/java,gradle,intellij
\ No newline at end of file
+# End of https://www.gitignore.io/api/java,gradle,intellij
+
+.idea/vcs.xml
+.idea/jarRepositories.xml
+.idea/compiler.xml
```
Comment (OlliKiljunen) [inline .gitignore]: I suggest we keep IDE config files out of the repo (even though this is an IDE project :D)

File: .travis.yml [hunk 2]
```diff
@@ -9,6 +9,7 @@ addons:
     token: $SONARCLOUD_TOKEN
 script:
   - sonar-scanner
+  - ./gradlew check
   
 notifications:
-  slack: letech:6iQNboxAjhcmAx2B0CW7LpJq
```
Comment (OlliKiljunen) [inline .travis.yml]: Travis showed alert concerning this so I put it in env var

File: build.gradle [hunk 3]
```diff
@@ -24,4 +25,9 @@ patchPluginXml {
     changeNotes """
       Add change notes here.<br>
       <em>most HTML tags may be used</em>"""
+}
+
+checkstyle {
+    configFile = file('/google_checks.xml')
+    maxWarnings = 0
```
Comment (OlliKiljunen) [inline build.gradle]: This disallows warnings i.e. minor crimes against style. Our code must be perfect(?)

File: .gitignore (hunk 1 above)
Comment (superseacat) [reply .gitignore]: oops!

File: .travis.yml (hunk 2 above)
Comment (superseacat) [reply .travis.yml]: regenerated the token in slack, added to travis

File: .gitignore (hunk 1 above)
Comment (OlliKiljunen) [reply .gitignore]: No, I mean, there were no such files in our repo. My IDE just generated some files that would had been part of this commit but I decided to put those in gitignore instead. Apparently it is arguable if files such as `.idea/misc.xml` should be in VCS or not but I think it's better not have them.

Comment (superseacat) [reply .gitignore]: ahhh... now I understand :) (I first thought I've pushed some "tech" files myself)

**Pull request merged**